src/EPR.Calculator.API/Controllers/CalculatorController.cs

```diff
@@ -35,84 +36,87 @@ public async Task<IActionResult> Create([FromBody] CreateCalculatorRunDto reques
-                    }
-
-                    // Read configuration items: message retry count and period
-                    var messageRetryCountFound = int.TryParse(this._configuration.GetSection("ServiceBus").GetSection("PostMessageRetryCount").Value, out int messageRetryCount);
-                    var messageRetryPeriodFound = int.TryParse(this._configuration.GetSection("ServiceBus").GetSection("PostMessageRetryPeriod").Value, out int messageRetryPeriod);
-
-                    if (!messageRetryCountFound)
-                    {
-                        return new ObjectResult("Configuration item not found: ServiceBus__PostMessageRetryCount") { StatusCode = StatusCodes.Status500InternalServerError };
-                    }
-
-                    if (!messageRetryPeriodFound)
+                    try
                     {
-                        return new ObjectResult("Configuration item not found: ServiceBus__PostMessageRetryPeriod") { StatusCode = StatusCodes.Status500InternalServerError };
-                    }
-
-                    // Get active default parameter settings master id
-                    var activeDefaultParameterSettingsMasterId = this.context.DefaultParameterSettings
-                        .SingleOrDefault(x => x.EffectiveTo == null && x.ParameterYear == request.FinancialYear)?.Id;
-
-                    // Get active lapcap data master id
-                    var activeLapcapDataMasterId = this.context.LapcapDataMaster
-                        .SingleOrDefault(data => data.ProjectionYear == request.FinancialYear && data.EffectiveTo == null)?.Id;
-
-                    // Setup calculator run details
-                    var calculatorRun = new CalculatorRun
-                    {
-                        Name = request.CalculatorRunName,
-                        Financial_Year = request.FinancialYear,
-                        CreatedBy = request.CreatedBy,
-                        CreatedAt = DateTime.Now,
-                        CalculatorRunClassificationId = (int)RunClassification.RUNNING,
-                        DefaultParameterSettingMasterId = activeDefaultParameterSettingsMasterId,
-                        LapcapDataMasterId = activeLapcapDataMasterId
-                    };
-
-                    // Save calculator run details to the database
-                    this.context.CalculatorRuns.Add(calculatorRun);
-                    this.context.SaveChanges();
-
-                    // Setup message
-                    var calculatorRunMessage = new CalculatorRunMessage
-                    {
-                        CalculatorRunId = calculatorRun.Id,
-                        FinancialYear = calculatorRun.Financial_Year,
-                        CreatedBy = User?.Identity?.Name ?? request.CreatedBy
-                    };
+                        // Return failed dependency error if at least one of the dependent data not available for the financial year
```

Comment:
My understanding is that your code changes should be in `DataPreChecksBeforeInitialisingCalculatorRun`
You do not need to touch this function.

Reply:
I thought why do we want to start transaction if it's running status that's why I kept on top but I'm okay if you want to change. I can write separate method for checking calculation already running inside 
```
using (var transaction = this.context.Database.BeginTransaction())
{ }
```
but I can't use the existing method `DataPreChecksBeforeInitialisingCalculatorRun `as it is specific to Status424FailedDependency error message.